drivers/ieee802154/ieee802154_gecko.c

```diff
@@ -0,0 +1,588 @@
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ieee802154_gecko, CONFIG_IEEE802154_DRIVER_LOG_LEVEL);
+
+#include <errno.h>
+#include <kernel.h>
+#include <arch/cpu.h>
+#include <debug/stack.h>
+#include <soc.h>
+#include <device.h>
+#include <init.h>
+#include <debug/stack.h>
+#include <net/net_if.h>
+#include <net/net_pkt.h>
+#include <sys/byteorder.h>
+#include <string.h>
+#include <random/rand32.h>
+#include <net/ieee802154_radio.h>
+
+#include <em_system.h>
+#include <rail.h>
+#include <rail_ieee802154.h>
+
+/* Defines maximum frame size used by the radio */
+#define RADIO_MAX_FRAME_SIZE                  128
+/* Maximum time to wait for RAIL to send the packet */
+#define TX_PACKET_SENT_TIMEOUT         K_MSEC(100)
+#define IEEE802154_FCS_LENGTH		        2
+
+struct ieee802154_gecko_context {
+	/* RAIL internal Tx FIFO */
+	u8_t rail_tx_fifo[RADIO_MAX_FRAME_SIZE] __aligned(RAIL_FIFO_ALIGNMENT);
+
+	RAIL_Handle_t rail_handle;
+
+	/* Pointer to the network interface. */
+	struct net_if *iface;
+
+	/* Device 802.15.4 long address. */
+	u8_t mac[8];
+
+	/* TX synchronization semaphore. Unlocked when frame has been sent or
+	 * send procedure failed.
+	 */
+	struct k_sem tx_wait;
+
+	/* TX result, updated in radio transmit callbacks. */
+	int tx_status;
+
+	/* Transmit / receive channel */
+	u16_t channel;
+};
+
+struct ieee802154_gecko_dev_cfg {
+	/* Initial TX power level in RAIL raw units */
+	RAIL_TxPowerLevel_t init_tx_power_level_raw;
+};
+
+#define DEV_NAME(dev) ((dev)->config->name)
+
+#define DEV_CFG(dev) \
+	((const struct ieee802154_gecko_dev_cfg *const)(dev)->config->config_info)
+
+#define DEV_DATA(dev) \
+	((struct ieee802154_gecko_context *const)(dev)->driver_data)
+
+static const RAIL_CsmaConfig_t rail_csma_config =
+	RAIL_CSMA_CONFIG_802_15_4_2003_2p4_GHz_OQPSK_CSMA;
+
+static const RAIL_LbtConfig_t rail_lbt_config = {
+	.lbtMinBoRand = 0,
+	.lbtMaxBoRand = 10,
+	.lbtTries = 5,
+	.lbtThreshold = -75,
+	.lbtBackoff = 320,   /* 20 symbols at 16 us/symbol */
+	.lbtDuration = 128,  /* 8 symbols at 16 us/symbol */
+	.lbtTimeout = 0,     /* No timeout */
+};
+
+static const RAIL_IEEE802154_Config_t rail_ieee802154_config = {
+	.addresses = NULL,
+	.ackConfig = {
+		.enable = false,
+		.ackTimeout = 672,
+		.rxTransitions = {
+			RAIL_RF_STATE_RX,
+			RAIL_RF_STATE_RX,
+		},
+		.txTransitions = {
+			RAIL_RF_STATE_RX,
+			RAIL_RF_STATE_RX,
+		},
+	},
+	.timings = {
+		.idleToRx = 100,
+		.idleToTx = 100,
+		.rxToTx = 192,
+		.txToRx = 192,
+		.rxSearchTimeout = 0,
+		.txToRxSearchTimeout = 0,
+	},
+	.framesMask = RAIL_IEEE802154_ACCEPT_STANDARD_FRAMES,
+	.promiscuousMode = false,
+	.isPanCoordinator = false,
+};
+
+static void ieee802154_gecko_rail_cb(RAIL_Handle_t rail_handle,
+				     RAIL_Events_t event);
+
+static RAIL_Config_t rail_config = {
+	.eventsCallback = &ieee802154_gecko_rail_cb,
+};
+
+static void ieee802154_gecko_get_eui64(u8_t *mac)
+{
+	u64_t uniqueID = SYSTEM_GetUnique();
+
+	memcpy(mac, &uniqueID, sizeof(uniqueID));
```

Comment:
Hi, it seems that from the EFR32xG13 reference manual there is a EUI48 register available from the DI (device information) maybe you can use it and convert it the value into a EUI-64.

Thinking of the Gecko Series 2, there is a EUI-64 value in the DEVINFO register. It will be nice to add a check whether it is a gecko series 1 or 2.
```suggestion
#ifdef _DEVINFO_EUI64L_MASK
sys_memcpy_swap(mac, (const uint8_t*)&DEVINFO->EUI64L, 8);
#elif _DEVINFO_EUI48L_MASK
uint8_t eui64[8];
uint8_t eui48[6];
memcpy(eui48, (const uint8_t*)&DEVINFO->EUI48L, 6);

eui64[7] = eui48[5];
eui64[6] = eui48[4];
eui64[5] = eui48[3];
eui64[4] = 0xFF;
eui64[3] = 0xFE;
eui64[2] = eui48[2];
eui64[1] = eui48[1];
eui64[0] = eui48[0];

sys_memcpy_swap(mac, eui64, 8);
#else
uint64_t uniqueID = SYSTEM_GetUnique();
memcpy(mac, &uniqueID, sizeof(uniqueID));
#endif
```